src/test/java/lesson9/SauceDemoCart_test.java

```diff
@@ -0,0 +1,65 @@
+package lesson9;
+
+import org.testng.annotations.*;
+import pageObjects.baseObjects.BaseTest;
+import pageObjects.saucedemo.BasketPage;
+import pageObjects.saucedemo.HeaderPage;
+import pageObjects.saucedemo.LoginPage;
+import pageObjects.saucedemo.ProductPage;
+import testNgUtils.Listener;
+
+@Listeners(Listener.class)
+public class SauceDemoCart_test extends BaseTest {
+
+    @Parameters("url")
+    @BeforeClass
+    public void preconditions(String url) {
+        new LoginPage()
+                .open(url)
```

Comment:
In order for the test to run successfully, need to use the methods to enter the login and password

    protected void enter(WebElement webElement, String enterData) {
        System.out.println("I'm enter :: " + enterData + ", by web element :: " + webElement);
        webElement.clear();
        webElement.sendKeys(enterData);
    }

because current method add 'a' symbol to login and password value

Reply:
 Я думаю, это происходит из-за комбинации клавиш для выделения текста внутри поля COMMAND+a (для MacOS), вместо CONTROL+a для других ос. (в классе BasePage)
` driver.findElement(locator).sendKeys(Keys.chord(Keys.COMMAND, "a", Keys.DELETE));` 
Для saucedemo эта строка не важна, но на других сайтах, где нужно почистить поле, control+a на MacOS не отрабатывает